app/frontend/src/features/profile/hooks/referencesHooks.ts

```diff
@@ -44,3 +61,66 @@ export function useReferencesReceived(
 
   return referencesReceivedQuery;
 }
+
+export const useListAvailableReferences = (userId: number) =>
+  useQuery<AvailableWriteReferencesRes.AsObject, GrpcError>(
+    availableWriteReferencesKey(userId),
+    () =>
+      service.references.getAvailableReferences({
+        userId,
+      })
+  );
+
+interface WriteHostRequestReferenceVariables {
+  referenceData: WriteHostRequestReferenceInput;
+}
+
+export function useWriteHostReference(userId: number) {
+  const queryClient = useQueryClient();
+  const {
+    mutate: writeHostRequestReference,
+    status,
+    reset,
+    error,
+    isLoading,
+  } = useMutation<
+    Reference.AsObject,
+    Error,
+    WriteHostRequestReferenceVariables
+  >(
+    ({ referenceData }) =>
+      service.references.writeHostRequestReference(referenceData),
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries([availableWriteReferencesKey(userId)]);
+      },
+    }
+  );
+
+  return { reset, status, writeHostRequestReference, error, isLoading };
+}
+
+interface WriteFriendReferenceVariables {
+  referenceData: WriteFriendReferenceInput;
+}
+
+export function useWriteFriendReference(userId: number) {
+  const queryClient = useQueryClient();
+  const {
+    mutate: writeFriendReference,
+    status,
+    reset,
+    error,
+    isLoading,
+  } = useMutation<Reference.AsObject, Error, WriteFriendReferenceVariables>(
+    ({ referenceData }) =>
+      service.references.writeFriendRequestReference(referenceData),
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries([availableWriteReferencesKey(userId)]);
```

Comment:
Same comment as above, especially this one as there won't be any waiting on the other party to write theirs first before the reference is allowed to show up